ECH new deployment API operations documented (#6277)
elastic/docs-content: deploy-manage/deploy/elastic-cloud/edit-stack-settings.md

For a list of supported settings, refer to the [{{es}} configuration reference](elasticsearch://reference/elasticsearch/configuration-reference/index.md). Settings supported on {{ech}} are indicated by an {{ecloud}} icon (![logo cloud](https://doc-icons.s3.us-east-2.amazonaws.com/logo_cloud.svg "Supported on {{ecloud}}")).

> ### Manage settings with the Cloud API

Reviewer: this change should also be exposed here: 

https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/6277/deploy-manage/stack-settings#configure-stack-settings

this makes me wonder: will these API improvements reach the ECE API in a future version?

Author: 🤷 :)

Author: About putting that content also in https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/6277/deploy-manage/stack-settings#configure-stack-settings I don't think it's extremely important. The ECH/ECE tab doesn't mention Cloud / ECE API at all and we have links at the end to the deployment-type-specific page where we are mentioning the API now.

If you want I can add an extra sentence saying that editing settings with the API is possible, and links to the ECE & ECH docs.

Reviewer: I think we should mention the API + link out at min 👍

Author: got it, I didn't realize the links to "this doc" are already there, we just need to mention that it covers API too.

Author: I've done a super minor change in the global stack-settings.md doc to address this. Let me know if this is what you mean.

> If you manage deployments through automation, use the dedicated deployment resource user settings endpoints for these single-purpose settings changes:

Reviewer: this sentence doesn't read exactly right. the use case (settings only) is kind of buried due to the sentence structure

```suggestion
If you manage deployments programmatically and only need to change your user settings, then you can use dedicated deployment resource user settings endpoints:
```

we also might want to expose that you can change settings as part of a larger deployment update?

- [Get the user settings of a Deployment resource]({{cloud-apis}}operation/operation-get-deployment-resource-user-settings)

> - [Update user settings for a deployment resource]({{cloud-apis}}operation/operation-update-deployment-resource-user-settings)

Reviewer: ```suggestion
- [Get the user settings of a deployment resource]({{cloud-apis}}operation/operation-get-deployment-resource-user-settings)
- [Update user settings for a deployment resource]({{cloud-apis}}operation/operation-update-deployment-resource-user-settings)
```

The capitalization on the targets looks funny, but we should just stylistically edit them so they look right over here

> 

Reviewer: I think this needs to be a tab or forked process on https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/6277/deploy-manage/deploy/elastic-cloud/edit-stack-settings#ec-add-user-settings, instead of being under "available settings"

Author: It was wrongly placed, sorry! I'll add it as a sub-section of Edit settings.

Author: fixed
